lib/note-content-editor.jsx

```diff
@@ -0,0 +1,68 @@
+import React, { PropTypes } from 'react';
+import { Editor, EditorState, ContentState } from 'draft-js';
+import { urlDecorator } from './url-decorator';
+
+function plainTextContent( editorState ) {
+	return editorState.getCurrentContent().getPlainText( '\n' )
+}
+
+export default React.createClass( {
+
+	propTypes: {
+		content: PropTypes.string.isRequired,
+		onChangeContent: PropTypes.func.isRequired
+	},
+
+	getInitialState() {
+		const { content } = this.props;
+		const contentState = ContentState.createFromText( content, '\n' );
+		return {
+			editorState: EditorState.createWithContent( contentState, urlDecorator )
+		}
+	},
+
+	handleEditorStateChange( editorState ) {
+		const contentHasChanged = plainTextContent( editorState )
+			!== plainTextContent( this.state.editorState )
+
+		this.setState( { editorState }, () => {
+			if ( contentHasChanged ) {
+				this.props.onChangeContent( plainTextContent( editorState ) );
+			}
+		} );
+	},
+
+	componentWillReceiveProps( nextProps ) {
+		if (
+			// if we've received this content before there's nothing to do now
+			nextProps.content !== this.props.content
+			// if the new content isn't different from what's rendered
+			// don't change it because we'd lose the SelectionState
+			&& plainTextContent( this.state.editorState ) !== nextProps.content
+		) {
+			const contentState = ContentState.createFromText( nextProps.content, '\n' );
+			this.setState( {
+				editorState: EditorState.createWithContent( contentState, urlDecorator )
+			} );
+		}
+	},
```

Comment:
(a 3rd situation, and this the only one where we really have to create a new `EditorState` is when the Editor starts rendering a different note because the user selected a different one in the sidebar, here we definitely don't want undo support)